Helper/Indexation.php

```diff
@@ -0,0 +1,84 @@
+<?php
+declare(strict_types=1);
+
+
+namespace Doofinder\Feed\Helper;
+
+use Magento\Framework\App\Helper\AbstractHelper;
+use Magento\Framework\App\Helper\Context;
+
+class Indexation extends AbstractHelper
+{
+
+    public const DOOFINDER_INDEX_PROCESS_STATUS_STARTED = 'STARTED';
+
+    public const DOOFINDER_INDEX_PROCESS_STATUS_SUCCESS = 'SUCCESS';
+
+    public const DOOFINDER_INDEX_PROCESS_STATUS_FAILURE = 'FAILURE';
+
+    public const DOOFINDER_GRID_SEVERITY_MINOR = 'minor';
+
+    public const DOOFINDER_GRID_SEVERITY_NOTICE = 'notice';
+
+    public const DOOFINDER_GRID_SEVERITY_MAJOR = 'major';
+
+    public const DOOFINDER_GRID_SEVERITY_CRITICAL = 'critical';
+
+    /**
+     * Indexation constructor.
+     *
+     * @param Context $context
+     */
+    public function __construct(Context $context) {
+        parent::__construct($context);
+    }
+
+    
+    /**
+     * Sanitize and prevent undefined index errors
+     *
+     * @param array $processTaskStatus
+     * @return array
+     */
+    public function sanitizeProcessTaskStatus(array $processTaskStatus): array
+    {
+        $status = $processTaskStatus['status'] ?? 'Unknown';
+        return [
+            'status'        => $status,
+            'result'        => $processTaskStatus['result'] ?? '',
+            'finished_at'   => $processTaskStatus['finished_at'] ?? '',
+            'error'         => $processTaskStatus['error'] ?? false,
+            'error_message' => isset($processTaskStatus['error_message']) ? $processTaskStatus['error_message'] : '',
+            'severity'      => $this->getSeverity($status)
+        ];
+    }
+
+    /**
+     * Get admin grid severity class by status
+     *
+     * @param string $status
+     * @return string
+     */
+    public function getSeverity(string $status): string
+    {
+        switch ($status) {
+            case self::DOOFINDER_INDEX_PROCESS_STATUS_STARTED:
+                $severity = self::DOOFINDER_GRID_SEVERITY_MINOR;
+                break;
+
+            case self::DOOFINDER_INDEX_PROCESS_STATUS_SUCCESS:
+                $severity = self::DOOFINDER_GRID_SEVERITY_NOTICE;
+                break;
+
+            case self::DOOFINDER_INDEX_PROCESS_STATUS_FAILURE:
+                $severity = self::DOOFINDER_GRID_SEVERITY_MAJOR;
+                break;
+
+            default:
+                $severity = self::DOOFINDER_GRID_SEVERITY_CRITICAL;
+                break;
+        }
+
+        return $severity;
+    }
+}
```

Comment:
Could you use a dict similar to this python code?
```python
severity = {
  DOOFINDER_INDEX_PROCESS_STATUS_STARTED: DOOFINDER_GRID_SEVERITY_MINOR,
  DOOFINDER_INDEX_PROCESS_STATUS_SUCCESS: DOOFINDER_GRID_SEVERITY_NOTICE
}
return severity.get(status, DOOFINDER_GRID_SEVERITY_CRITICAL)
```

Reply:
I've removed the "default" because with the new changes we won't have this case.